pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -175,18 +175,32 @@ func (p *DNSProxy) GetRules(endpointID uint16) restore.DNSRules {
 
 	restored := make(restore.DNSRules)
 	for port, entries := range p.allowed[uint64(endpointID)] {
+		count := 0
 		var ipRules restore.IPRules
 		for cs, regex := range entries {
 			var IPs map[string]struct{}
 			if !cs.IsWildcard() {
 				IPs = make(map[string]struct{})
+			Loop:
 				for _, nid := range cs.GetSelections() {
 					for _, ip := range p.LookupIPsBySecID(nid) {
 						IPs[ip] = struct{}{}
+						count++
+						if count > 1000 {
```

Comment:
Nice. Just curious how was 1000 chosen? Might be helpful to document why in the commit msg or similar for our future selves in case this ever comes back to bite us. (Unlikely, but worth spending the bit of extra effort :) )

Reply:
The concern here is that there would be a pathological policy that allows huge number of destinations (but not all, as the wildcard case is already optimized) and that would cause too much load as this thing is done every 5 seconds on each endpoint. Could have picked also 100, but felt that maybe some large loadbalanced DNS service might come too close to it. So basically 1000 felt like large enough to never cause trouble, while at the same time making sure we do not burn too much CPU doing this. WIll leave this like this for now having the good green CI run down there :-)